packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx

```diff
@@ -105,7 +109,7 @@ const lookupTokenIdConfigure = (
 
             let value: ContractTokenDetails[] = [];
 
-            if (!searchQuery) {
+            if (!searchQuery || !isHex(searchQuery) || !isEven(searchQuery.length)) {
```

Comment:
```suggestion
            if (!searchQuery || !isHex(searchQuery) || !isEven(searchQuery.length)) {
                setResult({ loading: false, value: undefined });
```

And then remove the conditional at line 179-181 (which does exactly the same, just prior to entering this function).

Reply:
:thinking: 
Why?

Reply:
The conditional at 179-181 only runs on `!searchQuery`, but the one marked above is more lax. As such, you could end in a place where the latest `setResult` call is `{loading: true, ...}`.

Either the 2 conditionals need to be aligned, or they should be consolidated as requested in my previous comment.